examples/chef/common/chef-dishwasher-mode-delegate-impl.h

```diff
@@ -0,0 +1,90 @@
+    ModeTagStructType modeTagsNormal[1] = { { .value = to_underlying(ModeTag::kNormal) } };
+    ModeTagStructType modeTagsHeavy[2]  = { { .value = to_underlying(ModeBase::ModeTag::kMax) },
+                                            { .value = to_underlying(ModeTag::kHeavy) } };
+    ModeTagStructType modeTagsLight[3]  = { { .value = to_underlying(ModeTag::kLight) },
+                                            { .value = to_underlying(ModeBase::ModeTag::kNight) },
+                                            { .value = to_underlying(ModeBase::ModeTag::kQuiet) } };
+
+    const detail::Structs::ModeOptionStruct::Type kModeOptions[3] = {
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Normal"),
+                                                 .mode     = ModeNormal,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(modeTagsNormal) },
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Heavy"),
+                                                 .mode     = ModeHeavy,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(modeTagsHeavy) },
+        detail::Structs::ModeOptionStruct::Type{ .label    = CharSpan::fromCharString("Light"),
+                                                 .mode     = ModeLight,
+                                                 .modeTags = DataModel::List<const ModeTagStructType>(modeTagsLight) }
+    };
+
+    CHIP_ERROR Init() override;
+    void HandleChangeToMode(uint8_t mode, ModeBase::Commands::ChangeToModeResponse::Type & response) override;
+
+    CHIP_ERROR GetModeLabelByIndex(uint8_t modeIndex, MutableCharSpan & label) override;
+    CHIP_ERROR GetModeValueByIndex(uint8_t modeIndex, uint8_t & value) override;
+    CHIP_ERROR GetModeTagsByIndex(uint8_t modeIndex, DataModel::List<ModeTagStructType> & tags) override;
+
+public:
+    ~DishwasherModeDelegate() override = default;
+};
+
+ModeBase::Instance * Instance();
+
+void Shutdown();
+
+} // namespace DishwasherMode
+
+} // namespace Clusters
+} // namespace app
+} // namespace chip
+
+#ifdef MATTER_DM_PLUGIN_DISHWASHER_MODE_SERVER
+chip::Protocols::InteractionModel::Status chefDishwasherModeWriteCallback(chip::EndpointId endpoint, chip::ClusterId clusterId,
+                                                                          const EmberAfAttributeMetadata * attributeMetadata,
+                                                                          uint8_t * buffer);
+chip::Protocols::InteractionModel::Status chefDishwasherModeReadCallback(chip::EndpointId endpoint, chip::ClusterId clusterId,
+                                                                         const EmberAfAttributeMetadata * attributeMetadata,
+                                                                         uint8_t * buffer, uint16_t maxReadLength);
+#endif
```

Comment:
endif needs comment indicating what what is the condition

Reply:
Done